### ISSUE TITLE
Minor incompatibilty with Matrix package leading to assignments by reference (':=') being printed

### ISSUE THREAD

**Issue by jan-glx** [Rdatatable/data.table#1277]
When the [`Matrix` package](http://r-forge.r-project.org/projects/matrix/) is loaded along with `data.table` (v1.9.5), assignments by reference using the `:=` operator in `j` lead to the resulting data.table being printed.
````
library(Matrix)
library(data.table)
dt=data.table(a=1)
dt[,b:=a]
````
````
> dt[,b:=a]
   a b
1: 1 1
````
Got me really confused (see profile pic) and I would love to see it fixed. 
best, 
Jan

**Comment by jangorecki** [comment]
Adding other potentially useful link: https://github.com/cran/Matrix/blob/master/R/zzz.R

**Comment by jan-glx** [comment]
I have no idea what that means (is it their fault/can data.table fix it?);
But it's great you are digging into it, @jangorecki!
Just wanted to note that the problem occurs with the [peplib package](https://cran.r-project.org/web/packages/peplib/index.html) as well.

best, Jan

**Comment by arunsrinivasan** [comment]
@jangorecki, great work! Do you mind testing if wrapping it with unique() would make this issue go away without failing other tests?

**Comment by arunsrinivasan** [comment]
Hm, I guess we have to check with older versions of R as well..

**Comment by arunsrinivasan** [comment]
Can't reproduce this with R v3.2.2. Please verify and reopen if you can still reproduce.

**Comment by jangorecki** [comment]
@arunsrinivasan I confirm, seems it has to be fixed just recently in R v3.2.2

**Comment by jan-glx** [comment]
Can confirm as well :)
